CSharp_Course_Project/Task3_Queue/StudentQueue.cs

```diff
@@ -0,0 +1,165 @@
+                var group = Console.ReadLine();
+
+                Console.Write("Please enter a student faculty: ");
+                var faculty = Console.ReadLine();
+
+                var student = new Student(name, surname, courseNumber, group, faculty);
+                this.students.Add($"{student.Name} {student.Surname}", student);
+
+                Console.WriteLine("Do you want to add one more student? Y/N");
+                var addResult = Console.ReadLine();
+                if (addResult.ToLower() == "y")
+                    continue;
+                else
+                    break;
+            }
+        }
+
+        public void RemoveStudent()
+        {
+            Console.WriteLine($"Choose one of the option to remove student(s)?{Environment.NewLine}" +
+                $"1 - Clear student list{Environment.NewLine}" +
+                $"2 - Remove student by name and surname");
+            var removeStatus = Console.ReadLine();
+            if (removeStatus == "1")
+                this.students.Clear();
+            else if (removeStatus == "2")
+            {
+                Console.Write("Enter student name: ");
+                var studentName = Console.ReadLine();
+                Console.Write("Enter student surname: ");
+                var studentSurname = Console.ReadLine();
+
+                students.Remove($"{studentName} {studentSurname}");
+            }
+            else
+                Console.WriteLine("Wrong option. Please, try again...");
+        }
+
+        public void ManageTask()
+        {
+            while (true)
+            {
+                Console.WriteLine($"Choose action:{Environment.NewLine}" +
+                                $"1 - Add task{Environment.NewLine}" +
+                                $"2 - Remove task{Environment.NewLine}" +
+                                $"3 - Print tasks{Environment.NewLine}" +
+                                $"4 - Assign tasks{Environment.NewLine}" +
+                                $"0 - Escape{Environment.NewLine}");
+                var action = Console.ReadLine();
+                switch (action)
+                {
+                    case "1":
+                        Console.Write("Add a task name: ");
+                        var taskName = Console.ReadLine();
+                        this.tasks.Enqueue(taskName);
+                        break;
+                    case "2":
+                        Console.WriteLine($"{this.tasks.Peek()} was removed");
+                        this.tasks.Dequeue();
+                        break;
+                    case "3":
+                        foreach (var task in this.tasks)
+                            Console.WriteLine(task);
+                        break;
+                    case "4":
+                        if (this.students.Count > 0 && this.tasks.Count > 0)
+                        {
+                            foreach (var value in this.students.Values)
```

Comment:
Please give appropriate name for student in dictionary.
It is not a value, it is a student.

Reply:
fixed